File: Zephiron/top-tier-customs/services/auth/src/routes/auth.routes.js
import express from "express";
import {
  getUser,
  health,
  login,
  logout,
  setCustomerStripeID,
  signup,
} from "../controllers/auth.controller.js";

const authRouter = express.Router();

authRouter.get("/health", health);
authRouter.get("/internal/users/:id", getUser);
authRouter.put("/internal/users/:id/stripe-customer", setCustomerStripeID);
authRouter.post("/signup", signup);
authRouter.post("/login", login);
authRouter.post("/logout", logout);

export default authRouter;

File: Projects/top-tier-customs/services/auth/internal/routes/auth_routes.go
package routes

import (
	"github.com/gin-gonic/gin"
	"github.com/samidmunir/top-tier-customs/services/auth/internal/handlers"
	"github.com/samidmunir/top-tier-customs/services/auth/internal/middleware"
)

func RegisterAuthRoutes(router *gin.Engine) {
	router.GET("/health", handlers.Health);
	router.POST("/signup", handlers.Signup);
	router.POST("/login", handlers.Login);
	router.POST("/logout", handlers.Logout);
	router.GET("/me", middleware.Authorize(), handlers.Me);
	router.POST("/refresh", handlers.Refresh);
}